test/lib/sdkdriver/src/dmp.ts

```diff
@@ -0,0 +1,132 @@
+import { EventEmitter } from 'events';
+import Transport from './transport';
+
+/**
+ * DMP Request.
+ * @class
+ * @classdesc A {@link DMPRequest} contains the data associated with
+ *   the incoming request message along with a method for sending a
+ *   response.
+ */
+class DMPRequest {
+  private readonly _id: number;
+  private readonly _transport: Transport;
+  public readonly data: any;
+
+  /**
+   * Constructor.
+   * @param {number} id - Request ID
+   * @param {*} data - Request data
+   * @param {Transport} transport - The {@link Transport} used for
+   *   sending a response
+   */
+  constructor(id: number, data: any, transport: Transport) {
```

Comment:
I believe you can just do

```
constructor(private readonly _id: number, public readonly data: any, private readonly transport: Transport) {}
```

however, then you don't get to have the nice parameter names—just the underscore prefixed ones. So I guess we ought to leave as-is.